core/tx_pool.go

```diff
@@ -583,7 +595,7 @@ func (pool *TxPool) add(tx *types.Transaction, local bool) (replaced bool, err e
 			return false, ErrUnderpriced
 		}
 		// New transaction is better than our worse ones, make room for it
-		drop := pool.priced.Discard(pool.all.Count()-int(pool.config.GlobalSlots+pool.config.GlobalQueue-1), pool.locals)
+		drop := pool.priced.Discard(pool.all.SlotsUsed()-int(pool.config.GlobalSlots+pool.config.GlobalQueue)+NumSlots(tx), pool.locals)
```

Comment:
If I'm reading this right, it means that one `4-slot` tx will push out `4` `1-slot` transactions, even though it's only marginally better. 
That may be fine, but please update the PR description with more information about the expected behaviours, such as this. 
These types of choices that is made internally deep within the tx pool determines whether the macro behaviour of the network can be attacked or not. 

It might be good to introduce some randomness to how the tx pool behaves in this regard.

Reply:
I've updated the PR description accordingly. Do you have concrete suggestions for the tx pool behavior?